homeassistant/components/tractive/sensor.py

```diff
@@ -0,0 +1,151 @@
+                    TractiveActivitySensor(
+                        client.user_id,
+                        trackable,
+                        tracker_details,
+                        unique_id,
+                        description,
+                    )
+                )
+
+    async_add_entities(entities)
+
+
+class TractiveSensor(SensorEntity):
+    """Tractive sensor."""
+
+    def __init__(self, user_id, trackable, tracker_details, unique_id, description):
+        """Initialize sensor entity."""
+        self._user_id = user_id
+        self._trackable = trackable
+        self._tracker_id = tracker_details["_id"]
+        self._attr_unique_id = unique_id
+        self._attr_device_info = {
+            "identifiers": {(DOMAIN, self._tracker_id)},
+            "name": f"Tractive ({self._tracker_id})",
+            "manufacturer": "Tractive GmbH",
+            "sw_version": tracker_details["fw_version"],
+            "model": tracker_details["model_number"],
+        }
+        self.entity_description = description
+
+    @callback
+    def handle_server_unavailable(self):
+        """Handle server unavailable."""
+        self._attr_available = False
+        self.async_write_ha_state()
+
+
+class TractiveHardwareSensor(TractiveSensor):
+    """Tractive hardware sensor."""
+
+    def __init__(self, user_id, trackable, tracker_details, unique_id, description):
+        """Initialize sensor entity."""
+        super().__init__(user_id, trackable, tracker_details, unique_id, description)
+        self._attr_name = f"{self._tracker_id} {description.name}"
+
+    @callback
+    def handle_hardware_status_update(self, event):
+        """Handle hardware status update."""
+        self._attr_state = event[self.entity_description.key]
+        self._attr_available = True
+        self.async_write_ha_state()
+
+    async def async_added_to_hass(self):
+        """Handle entity which will be added."""
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{TRACKER_HARDWARE_STATUS_UPDATED}-{self._tracker_id}",
+                self.handle_hardware_status_update,
+            )
+        )
+
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{SERVER_UNAVAILABLE}-{self._user_id}",
+                self.handle_server_unavailable,
+            )
+        )
+
+
+class TractiveActivitySensor(TractiveSensor):
+    """Tractive active sensor."""
+
+    def __init__(self, user_id, trackable, tracker_details, unique_id, description):
+        """Initialize sensor entity."""
+        super().__init__(user_id, trackable, tracker_details, unique_id, description)
+        self._attr_name = f"{trackable['details']['name']} {description.name}"
+
+    @callback
+    def handle_activity_status_update(self, event):
+        """Handle activity status update."""
+        self._attr_state = event[self.entity_description.key]
```

Comment:
```suggestion
        self._attr_native = event[self.entity_description.key]
```

Reply:
Since https://github.com/home-assistant/core/pull/48261 it's handled in the `SensorEntity` base class

https://github.com/home-assistant/core/blob/103e21c278300d328b1b5bbc78d9d6fd77908b6c/homeassistant/components/sensor/__init__.py#L158-L161

https://github.com/home-assistant/core/blob/103e21c278300d328b1b5bbc78d9d6fd77908b6c/homeassistant/components/sensor/__init__.py#L193-L236